camera_calibration/src/camera_calibration/camera_calibrator.py

```diff
@@ -132,6 +160,23 @@ def __init__(self, boards, service_check = True, synchronizer = message_filters.
         self._pattern = pattern
         self._camera_name = camera_name
         self._max_chessboard_speed = max_chessboard_speed
+        self._no_gui = no_gui
+        self._no_mono_if_stereo = no_mono_if_stereo
+
+        if self._no_mono_if_stereo:
+            left_camera_info_topic = "{}/camera_info".format(rospy.remap_name("left_camera"))
+            rospy.loginfo("Waiting for the CameraInfo msg from {} topic".format(left_camera_info_topic))
+            self.left_camera_info_msg = rospy.wait_for_message(left_camera_info_topic, sensor_msgs.msg.CameraInfo, 100)
+
+            right_camera_info_topic = "{}/camera_info".format(rospy.remap_name("right_camera"))
+            rospy.loginfo("Waiting for the CameraInfo msg from {} topic".format(right_camera_info_topic))
+            self.right_camera_info_msg = rospy.wait_for_message(right_camera_info_topic, sensor_msgs.msg.CameraInfo, 100)
+
+        if self._no_gui:
```

Comment:
i dont think that this is related to no gui. The publisher is an extra feature, maybe create an extra argument for it / remove it from the pr?

Reply:
If you have no_gui then you have no other visual feedback than those published images. I know you like non-topic approaches, but I would leave it for the next improvement, together with the verification process without Rviz.